sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -171,8 +172,7 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
             location = fileIndex,
             partitionSchema = partitionSchema,
             dataSchema = dataSchema,
-            // We don't support hive bucketed tables, only ones we write out.
-            bucketSpec = None,
+            bucketSpec = bucketSpec,
```

Comment:
I dont think this could cause problems (you can correct me if I am missing anything). Even if bucketing spec is set over here, this PR does not make use of that information in scan operators.